test/alternator/test_cql_rbac.py

```diff
@@ -0,0 +1,939 @@
+# Copyright 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# Tests for how CQL's Role-Based Access Control (RBAC) commands - CREATE ROLE,
+# GRANT, REVOKE, etc., can be used on Alternator for authentication and for
+# authorization. For example if the low-level name of an Alternator table "x"
+# is alternator_x.x, and a certain user is not granted permission to "modify"
+# keyspace alternator_x, Alternator write requests (PutItem, UpdateItem,
+# DeleteItem, BatchWriteItem) by that user will be denied.
+#
+# Because this file is all about testing the Scylla-only CQL-based RBAC,
+# all tests in this file are skipped when running against Amazon DynamoDB.
+
+import pytest
+import boto3
+from botocore.exceptions import ClientError
+import time
+from contextlib import contextmanager
+from test.alternator.util import is_aws, unique_table_name, random_string, new_test_table
+from functools import cache
+
+from cassandra.auth import PlainTextAuthProvider
+from cassandra.cluster import Cluster, ConsistencyLevel, ExecutionProfile, EXEC_PROFILE_DEFAULT
+from cassandra.policies import RoundRobinPolicy
+import re
+
+# This file is all about testing RBAC as configured via CQL, so we need to
+# connect to CQL to set these tests up. The "cql" fixture below enables that.
+# If we're not testing Scylla, or the CQL port is not available on the same
+# IP address as the Alternator IP address, a test using this fixture will
+# be skipped with a message about the CQL API not being available.
+@pytest.fixture(scope="module")
+def cql(dynamodb):
```

Comment:
As always I am a fan of unification versus duplication of code, we already have this code elsewhere. But I get that we don't agree on that :) so not insisting here, don't want to start the same discussion over and over.

Reply:
The lack of unification is what allowed me to write 1000 lines of functional test code this week instead of spending days battling with broken and unsuitable test frameworks.
In this case, this specific "cql" fixture is similar to the one in cql-pytest, but not identical. For example, it gets its configuration from the existing "dynamodb" fixture and not from some other means. It also doesn't go bonkers when faced with AWS but skips the test gentily. Since this is a 20 line function and not a 2000 line library, I think the "duplication" is not a problem. 

I agree that if this "cql" fixture becomes useful in additional test/alternator files, I can move it to conftest.py. We can and should definitely do it. But we CANNOT and SHOULD NOT share the cql fixture with test/cql-pytest or test/topology, and any attempt to do that will likely add in disastrous code.